pyvista/core/datasetattributes.py

```diff
@@ -1070,8 +1084,8 @@ def update(self, array_dict: Union[Dict[str, np.ndarray], 'DataSetAttributes']):
             self[name] = array.copy()
 
     def _raise_index_out_of_bounds(self, index: Any):
-        max_index = self.VTKObject.GetNumberOfArrays()
         if isinstance(index, int):
+            max_index = self.VTKObject.GetNumberOfArrays()
             if index < 0 or index >= self.VTKObject.GetNumberOfArrays():
```

Comment:
Originally editing to use `max_index`, but might as well make use of operator chaining:
```suggestion
            if not 0 <= index < max_index:
```